data-prepper-plugins/otel-trace-raw-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/oteltrace/OTelTraceRawProcessor.java

```diff
@@ -62,14 +62,13 @@ public OTelTraceRawProcessor(final OtelTraceRawProcessorConfig otelTraceRawProce
                                  final PluginMetrics pluginMetrics) {
         super(pluginMetrics);
         traceFlushInterval = SEC_TO_MILLIS * otelTraceRawProcessorConfig.getTraceFlushIntervalSeconds();
-        final int numProcessWorkers = pipelineDescription.getNumberOfProcessWorkers();
-        traceIdTraceGroupCache = CacheBuilder.newBuilder()
-                .concurrencyLevel(numProcessWorkers)
```

Comment:
Is there a similar setting in Caffeine that we should make use of here? In this case we know that there are only `numProcessWorkers` total modifying the cache, so this gives a hint to Guava on how to structure internally. How does Caffeine handle concurrency here?

Reply:
Caffeine does not accept such hints, as Guava Cache does, as it works differently under the hood.